docs/source/interfaces/utility/Fit Script Generator.rst

```diff
@@ -0,0 +1,204 @@
+
+**Generate Script to File**
+The data selected in the Fit Script Generator is used to generate a python script, which is then saved to a file.
+
+**Generate Script to Clipboard**
+The data selected in the Fit Script Generator is used to generate a python script, which is then copied to the clipboard. Paste this
+into a python script editor window.
+
+Concepts
+--------
+
+**Domain**
+This refers to a set of X, Y and error data at a specific index of a :ref:`MatrixWorkspace <MatrixWorkspace>`.
+
+**Function Index**
+Different domains can have the same function, and therefore have identically named parameters. A Function Index is used to differentiate
+between these identically named parameters by specifying the exact location of a parameter. For example if you are performing a simultaneous
+fit with two domains each containing a :ref:`FlatBackground <func-FlatBackground>`, the first domain will have a Function Index of `f0.`
+while the second domain has a Function Index of `f1.`. There are therefore two parameters, `f0.A0` and `f1.A0`.
+
+**Attribute**
+An attribute is a property of a function which doesn't change during fitting.
+
+**Parameter**
+A parameter is a decimal value property of a function which will change during fitting. It has an error.
+
+**Fix**
+A Fix is where a parameters value is forced to stay the same during a fit. It is a type of Tie.
+
+**Constraint**
+A Constraint is where a parameters value is forced to stay within a defined region during a fit.
+
+**Tie**
+A Tie is where a parameters value is set to be equal to the value of another parameter, or an expression, during a fit.
+
+Sequential Fitting Concepts
+---------------------------
+
+During a Sequential fit, each domain is fitted one after another. The parameter values at the end of a domain fit are
+passed to the next domain fit to be used as the initial parameter values.
+
+**Local Tie**
+A local tie is when you tie a parameter to another parameter which is in the same domain. This can be done for both
+sequential and simultaneous fitting.
+
+Simultaneous Fitting Concepts
+-----------------------------
+
+During a Simultaneous fit, each domain is fitted concurrently. This allows you to tie a parameter to another parameter
+in a different domain.
+
+**Global Tie**
+A global tie is when you tie a parameter to another parameter which is in a different domain. This can only be done for
+simultaneous fitting.
+
+**Global Parameter**
+A global parameter is a type of global tie. This is where each domain contains the same parameter, and it is 'shared' by
+each domain during a simultaneous fit.
+
+Usage Example for Sequential Fitting
+------------------------------------
+
+To carry out the following usage example you will first need to load some data into Mantid. Run the following script, and
+make sure you have access to the data archive.
+
+.. code-block:: python
+
+    from mantid.simpleapi import *
```

Comment:
Could we simply import what we need, e.g. `LoadMuonNexus, GroupWorkspace, RenameWorkspace, MuonPreProcess, MuonGroupingAsymmetry`

More because users tend to get in the habit of `import *` which (outside of Mantid especially) is considered poor practice